demos/demo-fake-smtp-server/app/Main.hs

```diff
@@ -53,12 +53,12 @@ sendSampleEmail smtpHostname smtpPort = do
 
   liftIO $ doSMTPFn $ \smtpConn -> do
     authSucceed <- if shouldAuth then authenticate PLAIN smtpUsername smtpPassword smtpConn else return True
-    case authSucceed of
-      True -> do
-        forM_ (mailTo mail) $ \(Address _ to) -> do
-          let Address _ from = mailFrom mail
-          sendMail mail smtpConn
-      False -> expectationFailure [i|Failed to authenticate to SMTP server #{smtpHostname} (port #{smtpPort}) with username #{smtpUsername}|]
+    if authSucceed then do
```

Comment:
This may be a personal thing but I hate this hlint warning, I don't like how `if/then/else` looks in code. Please remove, maybe we can add an hlint exception for this.

Better yet, we can eliminate the `authSucceed` variable and just do LambdaCase here